lib/perl/Genome/Model/Tools/Analysis/Coverage/CoveragePlot.pm

```diff
@@ -0,0 +1,1439 @@
+
+use Statistics::R;
+
+
+class Genome::Model::Tools::Analysis::Coverage::CoveragePlot{
+    is => 'Command',
+    has => [
+        roi_file_path => {
+	    is => 'String',
+	    is_optional => 0,
+            doc => 'Region-of-interest file in the BED format',
+        },
+	
+	output_directory => {
+	    is => 'String',
+	    is_optional => 0,
+	    doc => 'Directory to save output files such as PDF file and tables for plotting, and ref-cov stats files',
+	},
+	
+	bam_files => {
+	    is => 'String',
+	    is_optional => 1,
+	    doc => 'Comma-separated list of bam files to generate coverage plots from, Must specify either --bam_files or --refalign_models',
+	},
+	
+	refalign_models => {
+	    is => 'Number',
+	    is_optional => 1,
+	    doc => 'Comma-separated list of refalign model IDs to generate coverage plots from, bam_files will override refalign_models if given',
+	},
+	
+	labels => {
+	    is => 'String',
+	    is_optional => 1,
+	    doc => 'Comma-separated list of the names for each bam (or model), otherwise set such as 1, 2, 3, ... in the order',
+	},
+	
+        min_depth_filters => {
+            is => 'String',
+            doc => 'Comma-separated list of the minimum depths at each position to consider coverage plot, Must define at least 2 depths',
+            default_value => [1, 20, 40],
+            is_optional => 1,
+        },
+	
+        wingspan => {
+            is => 'Integer',
+            doc => 'A base pair wingspan value to add +/- of the input regions, Default is no wingspan',
+	    is_optional => 1,
+        },
+	
+	
+	# for plotting
+        bw => {
+            is => 'String',
+            doc => 'the smoothing bandwidth to be used in density, Default is "nrd0"',
+	    is_optional => 1,
+        },
+    ]
+};
+
+sub help_brief {
+    "Sequencing coverage analysis tool to create summary tables and coverage plots by runnning the ref-cov tool"
+}
+
+sub help_detail {
+    "Sequencing coverage analysis tool to create summary tables and coverage plots by runnning the ref-cov tool"
+}
+
+
+
+sub execute {
+    my $self = shift;
+    
+    #$DB::deep = 200;		# for debug
+    
+    
+    # defines the folder and file names
+    my $dir_refcov = "refcov";
+    my $filetab = "table.tsv";
+    my $script3 = catfile dirname(__FILE__), 'CoveragePlot.R';
+    
+    
+    # gets the arguments
+    my $roi_file_path = $self->roi_file_path;
+    my $output_directory = $self->output_directory;
+    
+    my $bam_files = $self->bam_files;
+    my $refalign_models = $self->refalign_models;
+    
+    my $labels = $self->labels;
+    my $min_depth_filters = $self->min_depth_filters;
+    my $wingspan = $self->wingspan;
+    
+    my $bw = $self->bw;
+    $bw = "nrd0" if ($bw =~ /^\s*$/);
+    
+    
+    # as default
+    die "roi-file-path required" unless -e $roi_file_path;
+    print STDERR sprintf("%d region(s) of interest from %s\n", `wc -l $roi_file_path | cut -d " " -f 1`, $roi_file_path);
```

Comment:
All these prints to STDERR should instead use the methods ```debug_message()```, ```status_message()```, ```warning_message()``` or ```error_message()``` depending on the severity of the message.  For example, this one could be:
```
$self->status_message("%d region(s) of interest from %s\n",
                      `wc -l $roi_file_path | cut -d " " -f 1`,
                      $roi_file_path);
```